src/jdk.compiler/share/classes/com/sun/tools/javac/model/JavacTypes.java

```diff
@@ -329,18 +329,17 @@ private void validateTypeNotIn(TypeMirror t, Set<TypeKind> invalidKinds) {
                 || elem.getModifiers().contains(Modifier.PRIVATE))
             return Collections.emptySet();
 
-        if (!(elem instanceof MethodSymbol))
+        if (!(elem instanceof MethodSymbol methodSymbol))
```

Comment:
These two look odd - your cleanup is fine, but it seems in the first cast, we want to wrap with IllegalArgumentException instead of failing with ClassCastException. In the second cast (for class symbol) it seems we're ok with just CCE. I'm fine with leaving this as per your cleanup - just noting the inconsistency.